src/commands/protocols/get/sender/get-assertion-merkle-root-command.js

```diff
@@ -15,19 +16,26 @@ class GetAssertionMerkleRootCommand extends Command {
      * @param command
      */
     async execute(command) {
-        const { operationId, blockchain, contract, knowledgeCollectionId } = command.data;
-        this.logger.info(
-            `Getting assertion id for token id: ${knowledgeCollectionId}, contract: ${contract}, operation id: ${operationId} on blockchain: ${blockchain}`,
-        );
+        const { operationId, ual } = command.data;
+        this.logger.info(`Getting assertion id and operation id ${operationId} for ual: ${ual}`);
 
-        const assertionId = await this.blockchainModuleManager.getLatestAssertionId(
-            blockchain,
-            contract,
-            knowledgeCollectionId,
+        let assertionId = await this.tripleStoreService.getLatestAssertionId(
+            TRIPLE_STORE_REPOSITORIES.PUBLIC_CURRENT,
+            ual,
         );
 
+        if (!assertionId) {
+            assertionId = await this.tripleStoreService.getLatestAssertionId(
+                TRIPLE_STORE_REPOSITORIES.PRIVATE_CURRENT,
+                ual,
+            );
+        }
+
+        if (!assertionId) {
+            throw new Error(`No assertionId found for UAL: ${ual} in either repository.`);
```

Comment:
Can we make this more descriptive maybe, we need to check what will client get from this